ConfigureSpicyBuild.cmake

```diff
@@ -0,0 +1,55 @@
+# Adapt the Spicy build to compiling it as part of Zeek, when some
+# defaults might be set differently than what Spicy expects.
+#
+# This is included from the top-level CMakelists.txt after the Spicy
+# subdirectory has been added.
+
+# We set custom compile options on the main Spicy targets.
+set(_spicy_targets
+    hilti-config
+    hilti-objects
+    hilti-rt-debug-objects
+    hilti-rt-objects
+    hiltic
+    jrx-objects
+    retest
+    spicy-batch-extract
+    spicy-config
+    spicy-driver
+    spicy-objects
+    spicy-rt-debug-objects
+    spicy-rt-objects
+    spicyc
+    testregex
+    spicy-doc
+    spicy-dump)
```

Comment:
I was wondering whether on could instead get these targets automatically, but it looks like e.g.,

```cmake
get_directory_property(_spicy_targets DIRECTORY auxil/spicy BUILDSYSTEM_TARGETS)
```

does not recurse, so probably good to just hardcode this for now. This would only become an issue if we changed the list of targets on the Spicy side.

Reply:
Yeah, thought the same: ok for now. We'll also catch any issues because basically this is just to get things to compile wo/ errors and warnings.